### ISSUE TITLE
patch module is not python 2.4 compatible

### ISSUE THREAD

**Issue by redbaron** [ansible/ansible#11453]
Issue Type:
Bug report

Ansible Version:
1.9.2

Environment:
CentOS 5.8 (Python 2.4)

Summary:
Patch module is incompatible with Python 2.4. It uses ternary operator which was introduced in Python 2.5 only.

Steps To Reproduce:
Just use patch module when remote host is running python 2.4

Expected Results:
Patch module works successfully

Actual Results:

```
failed: [remote_host] => (item={'key': 'nap_authenticate.patch', 'value': '/tmp/Authenticate.pm'}) => {"failed": true, "item": {"key": "nap_authenticate.patch", "value": "/tmp/Authenticate.pm"}, "parsed": false}
  File "/home/m.ivanov/.ansible/tmp/ansible-tmp-1435682288.08-149640426993369/patch", line 113
    msg = out if not err else err
               ^
```

**Comment by sivel** [comment]
# Closing This Ticket

Hi!

We believe recent commits (https://github.com/ansible/ansible-modules-extras/commit/e568032bc373b8328cc5faa81a3da0f893cb4c2d) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!

**Comment by redbaron** [comment]
I am sorry, but this is not the first time when fix is made and commited, but didn't make its way into a release.

Fix was made in May, latest 1.9.2 release was published in late Jun, and yet the problem exists.

You keep closing completely relevant tickets.

**Comment by bcoca** [comment]
@redbaron so the issue is fixed in current devel and upcomming 2.0, the reason it did not make 1.9.2 was because it was part of a large patch that affected many files, some of which are not present in 1.9 so we were not able to cherry pick the commit and decided to leave it out of the maintenance patch. 

Since this still seems to be a sore issue in your case I'll update the 1.9 with this specific change making sure it will be out with the 1.9.3 release.

**Comment by bcoca** [comment]
closing via 
https://github.com/ansible/ansible-modules-extras/commit/f8f2ba8c355e9be8576e733482835f11013e65ff
